### PR TITLE
Print stack trace in logical order

### PR DESCRIPTION
**Pull request by jglick**
Follow suggestion from [JDK-6507809](https://bugs.openjdk.java.net/browse/JDK-6507809) to print stack traces in a readable order.
- [x] handle exceptions which override `printStackTrace`
- [x] handle Java 7 suppressed exceptions
- [x] handle circular references
- [x] use system linefeed
- [x] complete unit tests
- [x] replace existing calls to `Throwable.printStackTrace`
- [X] use from `support-core`: https://github.com/jenkinsci/support-core-plugin/pull/96

@reviewbybees

### CODE REVIEW

**Comment by daniel-beck** [comment]
This would be *so* useful. @jglick Any plans to resurrect this PR?

**Comment by jglick** [comment]
Seems to cause test failures; some `TaskListener` loggers apparently treat `.print("foo\n")` differently from `.println("foo")`.

**Comment by jglick** [comment]
Indeed `PrintWriter.newLine()` does a `flush`, so these are not interchangeable. Will need to introduce convenience methods to replace original `printStackTrace` calls.

**Comment by jglick** [comment]
Random test failure.

**Comment by daniel-beck** [comment]
I don't think anyone cares as long as one of the two PR builds passes.

**Comment by daniel-beck** [comment]
@jglick Is this done and ready for review?

**Comment by jglick** [comment]
@daniel-beck yes I am awaiting review.

**Comment by stephenc** [comment]
:bee:

**Comment by jglick** [comment]
@reviewbybees done

File: core/src/main/java/hudson/Functions.java [hunk 1]
```diff
@@ -1438,19 +1443,87 @@ public static boolean defaultToTrue(Boolean b) {
...
+        }
+    }
+
+    /**
+     * Like {@link Throwable#printStackTrace(PrintWriter)} but using {@link #printThrowable} format.
+     * @param t an exception to print
+     * @param pw the log
+     * @since FIXME
+     */
+    public static void printStackTrace(@CheckForNull Throwable t, @Nonnull PrintWriter pw) {
```
Comment (daniel-beck) [inline core/src/main/java/hudson/Functions.java]: Name inconsistent with existing `printThrowable`, deliberate?

**Review by daniel-beck — APPROVED** [review]

File: core/src/main/java/hudson/Functions.java (hunk 1 above)
Comment (jglick) [inline core/src/main/java/hudson/Functions.java]: It is a replacement for an instance method in `Throwable`.

**Comment by stephenc** [comment]
:bee:

**Pull request merged**